Wrap possible NullPointerException

arthurscchan opened the pull request:
This PR provides a possible fix for issue [#124](https://github.com/FasterXML/jackson-datatypes-collections/issues/124) by adding null checking and wrapping possible NullPointerException thrown from underlying code or upstream libraries.

cowtowncoder commented on guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/GuavaCollectionDeserializer.java:
```diff
@@ -132,15 +132,19 @@ public Object deserializeWithType(JsonParser p, DeserializationContext ctxt,
+                return _deserializeFromSingleValue(p, ctxt);
+            }
+            return (T) ctxt.handleUnexpectedToken(_valueClass, p);
+        } catch (NullPointerException e) {
```
No. We'd need to know why NPE was thrown and prevent that instead -- a unit test reproduction would be useful.

arthurscchan replied:
I have added in unit test and also fix the logic of how to handle those NPE in https://github.com/FasterXML/jackson-datatypes-collections/pull/125/commits/3f3f61f66a43d0964693d74cd4aac3b513cdf7a1.

yawkat commented:
if you added the null check why do you need the try-catch still?

arthurscchan commented on guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/GuavaCollectionDeserializer.java:
```diff
@@ -132,15 +132,19 @@ public Object deserializeWithType(JsonParser p, DeserializationContext ctxt,
+            }
+            return (T) ctxt.handleUnexpectedToken(_valueClass, p);
+        } catch (NullPointerException e) {
+            throw new JsonProcessingException(e);
```
Have changed the logic and add in unit test in https://github.com/FasterXML/jackson-datatypes-collections/commit/3f3f61f66a43d0964693d74cd4aac3b513cdf7a1.

arthurscchan commented:
@yawkat Thanks for your comments. They are two fixes in different packages. But I have changed the logic to have null check in both cases and added in unit test.

cowtowncoder reviewed (approved):
Looks overall pretty good! I'll check out some details of null-handling but overall this works.

cowtowncoder commented:
Thank you @arthurscchan !
